### PR TITLE
Test improvements

### PR DESCRIPTION
**Pull request by orenbm**
#### What does this PR do?

1. Fix ssh helper for running command in remote machine - Fixes the way we close the connection. Also sends the password in the way suggested in CC.
2. Allow running cucumber tests with env_args without services - In the current implementation it was not possible to send env_args
without also sending services. In authn-azure we need to send env args
but we don't have any additional services so we need this ability.

### CODE REVIEW

File: cucumber/api/features/support/ssh_helpers.rb [hunk 1]
```diff
@@ -6,9 +6,9 @@
 module SshHelpers
 
   def run_command_in_machine(machine_ip, machine_username, machine_password, command)
-    ssh = Net::SSH.start(machine_ip, machine_username, :password => machine_password)
```
Comment (sigalsax) [inline cucumber/api/features/support/ssh_helpers.rb]: why was this changed? `:something` is a property of a hash that targets the key. Is NetSSH not expecting a password property?

**Comment by sigalsax** [comment]
NOTE: this is opened against master and not authn-azure

File: cucumber/api/features/support/ssh_helpers.rb (hunk 1 above)
Comment (orenbm) [reply cucumber/api/features/support/ssh_helpers.rb]: It does expect `password` and still gets it. I just changed the syntax

**Comment by orenbm** [comment]
> NOTE: this is opened against master and not authn-azure

@sigalsax it should be :)
